contrib/fslhd.py

```diff
@@ -257,8 +260,39 @@ def generate_nifti_fields(header):
         'sform_xorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[0]],
         'sform_yorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[1]],
         'sform_zorient': ORIENTATION_STRINGS[nib.aff2axcodes(sform)[2]],
-        'file_type': f"NIFTI-{int(nib_fields['magic'].decode('utf-8')[2])}+",
+        'file_type': f"NIFTI-{int(nib_fields['magic'].decode('utf-8', errors='ignore')[2])}+",
         'file_code': int(nib_fields['magic'].decode("utf-8")[2]),
-        'descrip': nib_fields['descrip'].decode('utf-8'),
-        'aux_file': nib_fields['aux_file'].decode('utf-8')
+        'descrip': nib_fields['descrip'].decode("utf-8", errors="ignore"),
+        'aux_file': nib_fields['aux_file'].decode("utf-8", errors="ignore")
+    }
+
+
+def generate_numpy_fields(arr):
+    """
+    Generate useful summary statistics for any arbitrary numpy array.
+    """
+    arr = arr.astype(np.float64)
+    return {
+        'arr_nbytes': arr.nbytes,
+        'arr_size': arr.size,
+        'arr_num_zeros': arr[arr == 0.0].size,
+        'arr_num_nonzeros': arr[arr != 0.0].size,
+        'arr_num_positives': arr[arr > 0.0].size,
+        'arr_num_negatives': arr[arr < 0.0].size,
```

Comment:
I think you can avoid some redundant array allocations by using `np.count_nonzero` instead:
```suggestion
        'arr_num_zeros': np.count_nonzero(arr == 0.0),
        'arr_num_nonzeros': np.count_nonzero(arr != 0.0),
        'arr_num_positives': np.count_nonzero(arr > 0.0),
        'arr_num_negatives': np.count_nonzero(arr < 0.0),
```

Reply:
It's really funny you came to this conclusion! I swear I had authored a commit that did exactly this. (I had stumbled across [this SO answer](https://stackoverflow.com/a/42916378) which is where I learned that `count_nonzero` count be used to efficiently count `== 0` elements!)

I don't know where that commit went, so thank you for bringing it up again. :stuck_out_tongue_closed_eyes: